### ISSUE TITLE
Django 1.11.1 build_attrs() got an unexpected keyword argument 'name'

### ISSUE THREAD

**Issue by rana-ahmed** [jdriscoll/django-photologue#175]
When I am using django-photologue with django version 1.11.1 I am getting this error 
**TypeError at /admin/photologue/gallery/add/**
build_attrs() got an unexpected keyword argument 'name'

**Comment by richardbarran** [comment]
Hi, could you post the whole traceback please?

**Comment by cristianjs19** [comment]
Hi Richard, the same has happened to me with django==1.11.1. Now I'm working with django==1.8 and the problem has disappear.
Here is the traceback:

```python
Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/admin/photologue/gallery/add/

Django Version: 1.11.1
Python Version: 2.7.12
Installed Applications:
('django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.sites',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.sitemaps',
 'photologue',
 'sortedm2m',
 'example_project')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware')


Template error:
In template /home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/contrib/admin/templates/admin/includes/fieldset.html, error at line 19
   build_attrs() got an unexpected keyword argument 'name'   9 :             {% for field in line %}
   10 :                 <div{% if not line.fields|length_is:'1' %} class="field-box{% if field.field.name %} field-{{ field.field.name }}{% endif %}{% if not field.is_readonly and field.errors %} errors{% endif %}{% if field.field.is_hidden %} hidden{% endif %}"{% elif field.is_checkbox %} class="checkbox-row"{% endif %}>
   11 :                     {% if not line.fields|length_is:'1' and not field.is_readonly %}{{ field.errors }}{% endif %}
   12 :                     {% if field.is_checkbox %}
   13 :                         {{ field.field }}{{ field.label_tag }}
   14 :                     {% else %}
   15 :                         {{ field.label_tag }}
   16 :                         {% if field.is_readonly %}
   17 :                             <div class="readonly">{{ field.contents }}</div>
   18 :                         {% else %}
   19 :                              {{ field.field }} 
   20 :                         {% endif %}
   21 :                     {% endif %}
   22 :                     {% if field.field.help_text %}
   23 :                         <div class="help">{{ field.field.help_text|safe }}</div>
   24 :                     {% endif %}
   25 :                 </div>
   26 :             {% endfor %}
   27 :         </div>
   28 :     {% endfor %}
   29 : </fieldset>


Traceback:

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/core/handlers/exception.py" in inner
  41.             response = get_response(request)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/core/handlers/base.py" in _legacy_get_response
  249.             response = self._get_response(request)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  217.                 response = self.process_exception_by_middleware(e, request)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  215.                 response = response.render()

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/response.py" in render
  107.             self.content = self.rendered_content

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/response.py" in rendered_content
  84.         content = template.render(context, self._request)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/backends/django.py" in render
  66.             return self.template.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  207.                     return self._render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/loader_tags.py" in render
  177.             return compiled_parent._render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/loader_tags.py" in render
  177.             return compiled_parent._render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/loader_tags.py" in render
  72.                 result = block.nodelist.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/loader_tags.py" in render
  72.                 result = block.nodelist.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/defaulttags.py" in render
  216.                     nodelist.append(node.render_annotated(context))

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/loader_tags.py" in render
  216.                 return template.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  209.                 return self._render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/defaulttags.py" in render
  216.                     nodelist.append(node.render_annotated(context))

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/defaulttags.py" in render
  216.                     nodelist.append(node.render_annotated(context))

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/defaulttags.py" in render
  322.                 return nodelist.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/defaulttags.py" in render
  322.                 return nodelist.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  990.                 bit = node.render_annotated(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_annotated
  957.             return self.render(context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render
  1046.         return render_value_in_context(output, context)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/template/base.py" in render_value_in_context
  1024.     value = force_text(value)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/utils/encoding.py" in force_text
  78.                 s = six.text_type(s)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/utils/html.py" in <lambda>
  376.         klass.__unicode__ = lambda self: mark_safe(klass_unicode(self))

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/forms/boundfield.py" in __str__
  41.         return self.as_widget()

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/forms/boundfield.py" in as_widget
  127.             **kwargs

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/forms/widgets.py" in render
  220.         context = self.get_context(name, value, attrs)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/django/contrib/admin/widgets.py" in get_context
  281.             'rendered_widget': self.widget.render(name, value, attrs),

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/sortedm2m/forms.py" in render
  44.         final_attrs = self.build_attrs(attrs, name=name)

File "/home/cristian/lenv/ph2/local/lib/python2.7/site-packages/sortedm2m/forms.py" in build_attrs
  35.             build_attrs(attrs, **kwargs)

Exception Type: TypeError at /admin/photologue/gallery/add/
Exception Value: build_attrs() got an unexpected keyword argument 'name'
```

**Comment by richardbarran** [comment]
Hi,
Thanks for the full traceback - the issue seems to be with sortedm2m, a package that Photologue depends on.
And [this issue has in fact already been reported on the sortedm2m project page](https://github.com/gregmuellegger/django-sortedm2m/issues/107); for now there's not much I can do apart from waiting for sortedm2m to fix and release a new version of their package.

**TL;DR**
For now, Photologue is not fully compatible with Django 1.11 :-)

**Comment by richardbarran** [comment]
The issue in django-sortedm2m has been fixed by a new release - I have updated the minimum version required by Photologue.
